### PR TITLE
Create and use unique petname paths

### PR DESCRIPTION
**Pull request by michaelfig**
Prevent crashing when a client suggests an existing path ("petname" that's an array of strings) for a different value.

#dapp-encouragement-branch: new-zoe

### CODE REVIEW

File: packages/dapp-svelte-wallet/api/src/lib-dehydrate.js
Comment (katelynsills) [inline packages/dapp-svelte-wallet/api/src/lib-dehydrate.js]: Hmm, there is something about this that violates the concept of petnames. I think there should not be a merge of autogenerated aspects + user-chosen names. The unknown-* defaults are the right way to deal with something that really has no name and no other information. @erights, what should be done?

**Review by katelynsills — CHANGES_REQUESTED** [review]
I think this might violate the concept of petnames, but I defer to @erights on that. Can we do something else, like if the petname already exists, the user cannot simply accept but must instead rename?

For example, how would a phone contacts list deal with this? I think they would reject if not unique, but would not automatically append anything.

File: packages/dapp-svelte-wallet/api/src/lib-dehydrate.js
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-dehydrate.js:240]: I can accept that for the string-based petnames.  For the array-based "paths", though, we already have a petname for the first element, but the rest of the elements are entirely chosen by the dapp.  It doesn't matter if we suffix them, they are already strong enough.

What is the correct behaviour in the pure petname (i.e. non-array-based) case?  The only client that would do this is one that isn't coming from a dapp, i.e. that already fully controls the ag-solo on which this code is running.

Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-dehydrate.js:240]: Correction: the dapp itself suggests a petname for the origin, which is used as the first component of all its paths.

**Review by dtribble — APPROVED** [review]
LGTM modulo the issue Kate raised. I think we should file that as a design question to revisit and get this in.

File: packages/dapp-svelte-wallet/api/src/lib-dehydrate.js
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-dehydrate.js]: @katelynsills would you PTAL?  I believe I've addressed your concern by only auto-suffixing petnames that are actually paths, since that is the only suggestion that can made over the wallet bridge by approved Dapps.

Unapproved Dapps can't suggest anything, and the only ones that can suggest bare strings are privileged users.

Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/lib-dehydrate.js]: I'm fine merging this in for now, but I think the "actually paths" distinction isn't a distinction. We aren't using edge names which are dynamic, so this is actually in the user's petname space whether we keep the path or not.

**Review by katelynsills — APPROVED** [review]

**Pull request merged**
